src/domain/Pool.ts

```diff
@@ -69,3 +69,55 @@ export function createPoolHandlers(addresses: PoolAddresses): void {
   juniorTokenContext.setString('tokenAddress', addresses.juniorToken)
   TokenTemplate.createWithContext(Address.fromString(addresses.juniorToken), juniorTokenContext)
 }
+
+export function createUpdatedPoolHandlers(prevAddresses: PoolAddresses, newAddresses: PoolAddresses): void {
+  let context = new DataSourceContext()
+  context.setString('id', newAddresses.id)
+
+  log.debug('createUpdatedPoolHandlers: {} => {}', [prevAddresses.id, newAddresses.id])
+
+  if (prevAddresses.coordinator != newAddresses.coordinator) {
+    log.debug('createUpdatedPoolHandlers: swapping coordinator {} => {}', [
+      prevAddresses.coordinator,
+      newAddresses.coordinator,
+    ])
+    CoordinatorTemplate.createWithContext(Address.fromString(newAddresses.coordinator), context)
```

Comment:
This does not swap them, it does only create the new handlers, right? Can we rename this log line and the ones below to: 

```suggestion
    log.debug('createUpdatedPoolHandlers: creating handler for changed coordinator {} => {}', [
      prevAddresses.coordinator,
      newAddresses.coordinator,
    ])
    CoordinatorTemplate.createWithContext(Address.fromString(newAddresses.coordinator), context)
```